packages/mobile/src/app/selectors.ts

```diff
@@ -108,6 +113,89 @@ export const linkBankAccountEnabledSelector = (state: RootState) => state.app.li
 
 export const sentryTracesSampleRateSelector = (state: RootState) => state.app.sentryTracesSampleRate
 
+export const supportedBiometryTypeSelector = (state: RootState) => state.app.supportedBiometryType
+
+export const biometryEnabledSelector = (state: RootState) =>
+  state.app.biometryEnabled && !!state.app.supportedBiometryType
+
+export const useBiometrySelector = (state: RootState) => state.app.useBiometry
+
+export const activeScreenSelector = (state: RootState) => state.app.activeScreen
+
 export const dappsListApiUrlSelector = (state: RootState) => state.app.dappListApiUrl
 
 export const superchargeButtonTypeSelector = (state: RootState) => state.app.superchargeButtonType
+
+type StoreWipeRecoveryScreens = Extract<
+  Screens,
+  | Screens.NameAndPicture
+  | Screens.ImportWallet
+  | Screens.VerificationEducationScreen
+  | Screens.VerificationInputScreen
+>
+type CreateAccountScreens = Extract<
+  Screens,
+  | Screens.NameAndPicture
+  | Screens.PincodeSet
+  | Screens.VerificationEducationScreen
+  | Screens.VerificationInputScreen
+>
+type RestoreAccountScreens = CreateAccountScreens & Screens.ImportWallet
+
+export const storeWipeRecoverySteps: { [key in StoreWipeRecoveryScreens]: number } = {
+  [Screens.NameAndPicture]: 1,
+  [Screens.ImportWallet]: 2,
+  [Screens.VerificationEducationScreen]: 3,
+  [Screens.VerificationInputScreen]: 3,
+}
+export const createAccountSteps: { [key in CreateAccountScreens]: number } = {
+  [Screens.NameAndPicture]: 1,
+  [Screens.PincodeSet]: 2,
+  [Screens.VerificationEducationScreen]: 3,
+  [Screens.VerificationInputScreen]: 3,
+}
+export const restoreAccountSteps: { [key in RestoreAccountScreens]: number } = {
+  [Screens.NameAndPicture]: 1,
+  [Screens.PincodeSet]: 2,
+  [Screens.ImportWallet]: 3,
+  [Screens.VerificationEducationScreen]: 4,
+  [Screens.VerificationInputScreen]: 4,
+}
+
+// The logic in this selector should be moved to a hook when all registration
+// screens are function components
+export const registrationStepsSelector = createSelector(
+  [
+    choseToRestoreAccountSelector,
+    biometryEnabledSelector,
+    activeScreenSelector,
+    recoveringFromStoreWipeSelector,
+  ],
+  (chooseRestoreAccount, biometryEnabled, activeScreen, recoveringFromStoreWipe) => {
+    let step = 0
+    let totalSteps = 3
```

Comment:
Shouldn't the base `totalSteps` be `4` instead if the user is restoring an account?

Reply:
yes if the user is restoring account i have a `totalSteps++` in the `if (chooseRestoreAccount)` block...but i will make this less confusing by making the separate blocks